### PR TITLE
Mat 8131 fixing timing of validation

### PR DESCRIPTION
**Pull request by gregory-akins**
## MADiE PR

Jira Ticket: [MAT-8131](https://jira.cms.gov/browse/MAT-8131)
(Optional) Related Tickets:

### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.test.tsx [hunk 1]
```diff
@@ -474,6 +477,34 @@ describe("TypeEditor Component", () => {
...
+        />
+      </FormikProvider>
+    );
+    const inputField = screen.getByTestId(
+      "uri-input-field-DiagnosticReport.presentedForm.uri"
+    );
+    expect(inputField).toBeInTheDocument();
+    await act(async () => {
+      userEvent.type(inputField, "urn:oid:AA");
+    });
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.test.tsx:504]: Seems this pattern of `await act()` is deprecated, recommend replacing with `waitFor`:
```suggestion
    await waitFor(async () => {
      userEvent.type(inputField, "urn:oid:AA");
    });
```

Comment (gregory-akins) [inline src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.test.tsx:504]: In this case, I think I'm using "act" correctly because it's to ensure that the userEvent processes correctly
Using this as a reference
https://rathoreaparna678.medium.com/act-and-waitfor-react-testing-library-49dec87acb48

Comment (jkotanchik-SB) [inline src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.test.tsx:504]: In another test file I see `act` is deprecated, maybe change to `waitFor`?

Comment (jkotanchik-SB) [reply src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.test.tsx:504]: Never mind, I'm still using the old import in my test file. Using  act in "testing-library/react" resolves the deprecation warning in my local.

**Review by jkotanchik-SB — APPROVED** [review]
